### PR TITLE
Webpack v4 addendum

### PR DESCRIPTION
**Pull request by julienben**
# Update Log

This is a follow up to the Webpack v4 upgrade thread [here](https://github.com/react-boilerplate/react-boilerplate/pull/2148).

The PWA manifest.json is now generated by the webpack-pwa-manifest plugin. I updated the documentation accordingly and also upgraded some dependencies.

I also disabled JPEG optimization via image-webpack-loader as it was breaking the build in Travis CI (no matter the version). I left a NOTE explaining how to enable it quickly.

## Tooling Versions

* Node 8.9.4
* npm 5.7.1
* yarn 1.5.1
* Mac OS 10.13.3
* Chrome 65.0.3325.162 (64-bit)

## :package: Version Diff

**[0] PATCH UPDATES**

```
 hoist-non-react-statics                    2.3.1  →  2.5.0
 enzyme                                     3.2.0  →  3.3.0
 stylelint-config-standard                  18.0.0 → 18.2.0
 stylelint-processor-styled-components      1.1.1  →  1.3.1
```

**[1] MINOR UPDATES**

```
 chalk                                   2.3.0  →    2.3.2
 compression                             1.7.1  →    1.7.2
 cross-env                               5.1.1  →    5.1.4
 express                                4.16.2  →   4.16.3
 invariant                               2.2.2  →    2.2.4
 lodash                                 4.17.4  →   4.17.5
 prop-types                             15.6.0  →   15.6.1
 react-redux                             5.0.6  →    5.0.7

 css-loader                            0.28.10  →  0.28.11
 enzyme-adapter-react-16                 1.1.0  →    1.1.1
 eslint-import-resolver-webpack          0.8.3  →    0.8.4
 html-webpack-plugin                     3.0.4  →    3.0.6
 rimraf                                  2.6.1  →    2.6.2
 style-loader                           0.20.2  →   0.20.3
 webpack                                 4.1.0  →    4.1.1
 webpack-cli                            2.0.10  →   2.0.12
 webpack-dev-middleware                  3.0.0  →    3.0.1
```

**[3] MAJOR UPDATES**

```
 image-webpack-loader                  3.6.0  →   4.2.0
```

**[4] NEW DEPENDENCIES**

```
 webpack-pwa-manifest                           3.6.2
```

### Warnings: Observed on `yarn install` after removing node_modules

*  warning "stylelint > postcss-html@0.11.0" has unmet peer dependency "postcss-sass@>=0.2.0".

### CODE REVIEW

**Comment by Mensae** [comment]
Looks good @julienben 

- Node 8.10.0
- npm 5.6.0
- yarn 1.5.1
- Windows 10.0.16299.309
- Chrome 65.0.3325.181 (64-bit)

I would've liked it even better if you'd upgraded styled-components too :-)

```
styled-components 2.3.0 → 3.2.5
```

I'm using it myself for a while now.

**Comment by julienben** [comment]
This was mostly meant to improve on the changes in my Webpack 4 PR. I believe there's another PR with more dependency upgrades.

@gretzky Are you back and able to review this by any chance? I'd like to contribute more and help get the repo ready for another big release but it feels a little slow...

**Comment by gretzky** [comment]
Hey @julienben, indeed it is a little slow and that's my bad. I got unexpectedly swamped at work the last couple of months, but I really appreciate the work you've put in and want to make sure I block off some time to take a look. If you're interested in becoming a member, I'm sure @mxstbr would love to have you!

**Comment by julienben** [comment]
Hey @gretzky 

Thanks! I would be very happy to become a member. Anything I need to do to make that happen?

More related to this PR: I just updated webpack to 4.4.1 and made more performance upgrade based on this [new guide](https://developers.google.com/web/fundamentals/performance/webpack/) from Google.

Julien

**Comment by gretzky** [comment]
@julienben, head over to Spectrum and [reply to this thread](https://spectrum.chat/?t=c22f1083-f693-47a2-9920-61495a2a4947)

File: app/app.js [hunk 1]
```diff
@@ -23,23 +23,9 @@ import App from 'containers/App';
 // Import Language Provider
 import LanguageProvider from 'containers/LanguageProvider';
 
-// Load the favicon, the manifest.json file and the .htaccess file
-/* eslint-disable import/no-webpack-loader-syntax */
+// Load the favicon and the .htaccess file
```
Comment (gretzky) [inline app/app.js]: Why'd we remove the multi-sized icons?

File: app/components/LoadingIndicator/Circle.js [hunk 2]
```diff
@@ -2,6 +2,8 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import styled, { keyframes } from 'styled-components';
 
+/* stylelint-disable comment-empty-line-before */
```
Comment (gretzky) [inline app/components/LoadingIndicator/Circle.js]: Is there a stylelint config? (sorry if I haven't checked the dev branch and there is)

File: internals/webpack/webpack.prod.babel.js [hunk 3]
```diff
@@ -72,6 +77,26 @@ module.exports = require('./webpack.base.babel')({
 
       AppCache: false,
     }),
+
```
Comment (gretzky) [inline internals/webpack/webpack.prod.babel.js]: This should be its own PR, it's larger than just a Webpack4 update

File: app/app.js (hunk 1 above)
Comment (julienben) [reply app/app.js]: We didn't. We keep only the largest icon (512*512) and, on build, webpack-pwa-manifest generates all of the other icons for us. They're automatically added in the build folder and linked to in the generated manifest.json.

Cf line 70 of webpack.prod.babel.js to see what I mean.

File: app/components/LoadingIndicator/Circle.js (hunk 2 above)
Comment (julienben) [reply app/components/LoadingIndicator/Circle.js]: Yup it's in dev. You merged it in PR #2066 ;)

File: internals/webpack/webpack.prod.babel.js (hunk 3 above)
Comment (julienben) [reply internals/webpack/webpack.prod.babel.js]: You're talking about webpack-pwa-manifest? It's basically the reason for this PR and for almost all of the changes here... I wrote it in response to the post-merge comments on my webpack 4 PR (#2148). Perhaps my mistake was naming the PR "addendum" but it has almost no reason to be without webpack-pwa-manifest.

If you really want me to break it into 2 I would rather remove from this one everything which isn't related to webpack-pwa-manifest (small dependency upgrades and bundle size improvements). Lmk what you think!

File: app/components/LoadingIndicator/Circle.js (hunk 2 above)
Comment (gretzky) [reply app/components/LoadingIndicator/Circle.js]: Thank you 😅

File: internals/webpack/webpack.prod.babel.js (hunk 3 above)
Comment (gretzky) [reply internals/webpack/webpack.prod.babel.js]: Got it, I messed with it a bit more and I agree now with why its in this PR. Disregard

**Comment by gretzky** [comment]
wtb @julienben !

File: app/app.js (hunk 1 above)
Comment (gretzky) [reply app/app.js]: Got it, I messed with it a bit more and I agree now with why its in this PR. Disregard

**Comment by Mensae** [comment]
Hi @julienben 

I'm trying this for the first time with some SVG and notice that you forgot to add ```svg-url-loader``` to ```package.json```

**Comment by julienben** [comment]
🤦🤦🤦

#2185
